workers/default/xml/xml_validator.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+from lxml import etree
+
+log = logging.getLogger(__name__)
+
+
+def validate_xml(xml_file_path, schema_file_path=None):
```

Comment:
Ich würde entweder ein boolean Result zurückgeben oder die Funktion in `assert_xml_valid` umbenennen. Dann könnte man am Namen schon ablesen, wie die Methode zu nutzen ist.

Außerdem sollten die Exceptions, die geworfen werden können irgendwo dokumentiert werden (Schlüsselwort `raises`).

Man könnte sich zusätzlich überlegen, die Exceptions in eigene Klassen zu wrappen, damit sich der Typ nicht ändert, sollte sich die interne Implementation der Klasse ändern, in dem z.B. eine andere XML-Library eingesetzt wird.

Reply:
Also bei der Exception-Doku gehe ich mit. Müssten wir uns nur einigen wie das genau zu machen ist, weil wir über sowas afair noch nicht gesprochen haben.

Was Benamung angeht, finde ich die eigentlich ok so. Wenn ich da bool zurückgebe, müsste ich die "isValid" oder so nennen. Da finde ich ich validate schon besser. Assert assoziiere ich irgendwie (unit)Tests. Andererseits scheint dein Vorschlag auch eher dem Vorgehen in der lxml library zu entsprechen.
Wenn du darauf bestehst, füge ich mich, ansonsten brauchen wir da mal eine dritte Meinung. :P

Exceptions wrappen finde ich grundsätzlich keine schlechte Idee, würde ich aber in die Kategorie over-engineering einordnen. :)

Reply:
Ich finde es eigentlich okay so, tut ja was es soll. Wrappen würde das ganze halt etwas übersichtlicher machen (und ggf. robuster). Würde Richard hier aber zustimmen und das wrappen als over engineered bezeichnen. lxml übernimmt das ja für uns.

Reply:
OK, Wrappen ist nicht nötig. Kann man sich überlegen, wenn wir das Exception-Handling insgesamt mal vereinheitlichen wollen, aber da haben wir definitiv akutere Probleme ...

Was den Methodennamen angeht: Ich bin da glaube ich auch noch sehr durch Java geprägt, wo das Credo ist "only use exceptions for exceptional states". Und prinzipiell gehe ich, da wir versuchen, einen funktionalen Stil zu nutzen, davon aus, dass der primäre Zweck einer Methode die Berechnung eines Rückgabewerts ist. Der Präfix `assert` würde deutlich machen, dass eine Exception hier erwartetes Verhalten ist, man quasi explizit vom Standardverhalten abweicht.

Ich will allerdings auch nicht drauf bestehen, dafür ist es auch zu unwichtig ;)